bpo-14243: Optionally delete NamedTemporaryFile on content manager exit but not on file close (#22431)
python/cpython: Doc/whatsnew/3.12.rst

Other Language Changes
======================

> * The :class:`tempfile.NamedTemporaryFile` has a new optional parameter *delete_on_close* (Contributed by Evgeny Zorin in :gh:`22431`.)

Reviewer: ```suggestion
* The :class:`tempfile.NamedTemporaryFile` has a new optional parameter *delete_on_close* (Contributed by Evgeny Zorin in :issue:`14243`.)
```

We typically refer to the issue number, not the pull request. That should also get around the (incorrect) error here.

Reviewer 2: Sorry, copy and paste error - I meant to suggest updating the bpo `:issue:` role to GitHub -- the number should be #58451

```suggestion
* The :class:`tempfile.NamedTemporaryFile` has a new optional parameter *delete_on_close* (Contributed by Evgeny Zorin in :gh:`58451`.)
```

A

> * XXX The :class:`tempfile.NamedTemporaryFile` has a new optional parameter *delete_on_close* (Contributed by Evgeny Zorin in :issue:`14243`.)

Reviewer 2: This should go in Improved Modules -> tempfile
```suggestion
* The :class:`tempfile.NamedTemporaryFile` has a new optional parameter *delete_on_close* (Contributed by Evgeny Zorin in :gh:`22431`.)
```

Reviewer 2: @Ev2geny -- you updated the text but you need to move it to the section I mentioned (see https://github.com/python/cpython/blob/5893b5db98b38b17750c0572c7209774a5034898/Doc/whatsnew/3.9.rst?plain=1#L327-L332) for what the section should look like -- you'll need to add `tempfile` as it's the first improvement listed in whatsnew.

A

Author: @AA-Turner , thanks. i have fixed it now.

Improved Modules
================

tempfile
--------

> The :class:`tempfile.NamedTemporaryFile` has a new optional parameter *delete_on_close* (Contributed by Evgeny Zorin in :gh:`58451`.)

Reviewer 2: Wrap lines at 70-80 chars
```suggestion
The :class:`tempfile.NamedTemporaryFile` has a new optional parameter 
*delete_on_close* (Contributed by Evgeny Zorin in :gh:`58451`.)
```

Improved Modules
================

tempfile
--------

> The :class:`tempfile.NamedTemporaryFile` has a new optional parameter 

Reviewer 3: ```suggestion
The :class:`tempfile.NamedTemporaryFile` class has a new optional parameter 
```
